File: server/src/main/java/com/nbu/projects/dentistappointmentsys/controllers/request_models/register/DentistRegisterModel.java
package com.nbu.projects.dentistappointmentsys.controllers.request_models.register;

import com.nbu.projects.dentistappointmentsys.models.types.DentistType;
import com.nbu.projects.dentistappointmentsys.models.types.Role;

public class DentistRegisterModel extends UserRegisterModel {

  String city;
  DentistType dentistType;
  String generalInformation;

  public DentistRegisterModel() {
  }

  public DentistRegisterModel(Long Id,
                              String email,
                              String password,
                              String firstName,
                              String lastName,
                              Role role,
                              String city,
                              DentistType dentistType,
                              String generalInformation) {
    super(Id,email, password, firstName, lastName, role, city, dentistType,generalInformation);
    this.city = city;
    this.dentistType = dentistType;
  }

  public String getCity() {
    return city;
  }

  public void setCity(String city) {
    this.city = city;
  }

  public DentistType getDentistType() {
    return dentistType;
  }

  @Override
  public String getGeneralInformation() {
    return generalInformation;
  }

  @Override
  public void setGeneralInformation(String generalInformation) {
    this.generalInformation = generalInformation;
  }

  public void setDentistType(DentistType dentistType) {
    this.dentistType = dentistType;
  }
}

File: ui/src/main/ui/src/app/main/public/register/model/request/dentist.register.model.ts
import {RegisterModel} from "../register.model";
import {Constants} from "../../../../../models/constants";

export class DentistRegisterModel {
   email: string;
   password: string;
   firstName: string;
   lastName: string;
   role: string;

   city: string;
   dentistType: string;

   constructor(registerModel: RegisterModel) {
      this.email = registerModel.email;
      this.password = registerModel.password;
      this.firstName = registerModel.firstName;
      this.lastName = registerModel.lastName;
      this.role = Constants.ROLE_DENTIST;
      this.city = registerModel.city;

      switch (registerModel.type) {
         case Constants.TYPE_ORTHODONT:
            this.dentistType = Constants.TYPE_ORTHODONT;
            break;
         case Constants.TYPE_SURGEON:
            this.dentistType = Constants.TYPE_SURGEON;
            break;
         case Constants.TYPE_NORMAL:
         default:
            this.dentistType = Constants.TYPE_NORMAL;
      }
   }
}
